File: backend/Polyglot.DataAccess/Entities/TeamTranslator.cs
using System.Collections.Generic;

namespace Polyglot.DataAccess.Entities
{
    public class TeamTranslator : Entity
    {
        public int TeamId { get; set; }
        public virtual Team Team { get; set; }

        public int TranslatorId { get; set; }

        public virtual UserProfile UserProfile { get; set; }

        public bool IsActivated { get; set; }

        public virtual ICollection<TranslatorRight> TranslatorRights { get; set; }

        public TeamTranslator()
        {
            IsActivated = false;
            TranslatorRights = new List<TranslatorRight>();
        }
    }
}

File: frontend/src/app/models/team-translator.ts
import { Team } from "./team";
import { TranslatorRight } from "./translator-right";
import { UserProfile } from ".";

export interface TeamTranslator {
    id: number;
    teamId: number;
    team: Team;
    translatorId: number;
    translator: UserProfile;
    translatorRights: Array<TranslatorRight>;
}
